src/query_parser/parser.y

```diff
@@ -260,7 +268,27 @@ expr(A) ::= expr(B) expr(C) . [AND] {
     }
 }
 
-text_expr(A) ::= text_expr(B) text_expr(C) . [AND] {
+// This rule is needed for queries like "hello (world @loc:[15.65 -15.65 30 ft])", when we descover too late that
```

Comment:
```suggestion
// This rule is needed for queries like "hello (world @loc:[15.65 -15.65 30 ft])", when we discover too late that
```

Reply:
So each left parenthesis is increasing the stack size.
Maybe we can consider increasing the stack size using the [`%stack_size` directive](https://sqlite.org/src/doc/trunk/doc/lemon.html#stack_size) (default size is 100)
@meir @ashtul @rafie WDYT?